crates/batch-prover/src/proving.rs

```diff
@@ -201,8 +202,18 @@ where
         }
     }
 
+    let last_l2_height = sequencer_commitments
+        .last()
+        .expect("Should have at least 1 commitment")
+        .l2_end_block_number;
+    let current_spec = fork_from_block_number(FORKS, last_l2_height).spec_id;
+    let elf = elfs_by_spec
+        .get(&current_spec)
+        .expect("Every fork should have an elf attached")
+        .clone();
+
```

Comment:
the grouping that'll be applied to these sequencer commitments are important.

let's say there is 3 items i n`sequencer_commmitments` vec, and only the last one has the height which the next fork activates.

down in execution, if these guys are grouped like [(1,2), (3)] than we can prove the first group  elf n-1 and the last group with elf 3. 

or else there will be confusion when we need to pull batch proof method ids based on `output.last_l2_height` in light client circuit

Reply:
I have implemented a change in `break_sequencer_commitments` to not only account for state diff threshold, but also forks. Commitments will be grouped by fork.

Reply:
One thing to note, is that with:
`GroupCommitment::SingleShot`: we don't account for forks
`GroupCommitment::OneByOne`: Each commitment is split by default so no need to account for forks
`break_sequencer_commitment`: Accounts for forks.